config/config.go

```diff
@@ -111,15 +111,6 @@ func (cfg *AnalyzerConfig) Validate() error {
 	if cfg.Name == "" {
 		return fmt.Errorf("malformed analyzer name '%s'", cfg.Name)
 	}
-	if cfg.ChainID == "" {
-		return fmt.Errorf("malformed chain id '%s'", cfg.ChainID)
-	}
-	if cfg.RPC == "" {
-		return fmt.Errorf("malformed RPC endpoint '%s'", cfg.RPC)
-	}
-	if cfg.ChainContext == "" {
-		return fmt.Errorf("malformed chain context '%s'", cfg.ChainContext)
-	}
```

Comment:
Looks like these are still needed for the consensus/emerald analyzers; do we want to move this logic to `[consensus/emerald].NewMain`? Or maybe the returned error when trying to connect w/ an empty context is descriptive enough?

Reply:
Yea good catch. This is not a great change. The config struct is a bit tech-debty now.

It used to describe analyzers that had to do with scanning a chain block by block. With the new analyzers, the set of analyzers is more diverse. Really we'd need 2+ different config objects, and then we'd need to refactor the whole config file to accommodate that. It would be cleaner and better, but I didn't want to spend time on it. My reasoning was that 99%-100% of the time, we're tweaking existing configs, not creating them from scratch. And those "templates" provide stronger guidance than these removed checks will.